### DISCUSSION TITLE
Use global variables for component functions

### DISCUSSION THREAD

**Discussion by wanjakarl** [flyntwp/flynt#365] [Q&A] [answered]
Is it possible to use variables like this?

**Your snippet**: https://github.com/flyntwp/flynt#advanced-custom-fields

**My snippet**:

```php
<?php

namespace Flynt\Components\BlockWysiwyg;

$var = "foo";

function getACFLayout()
{
    global $var;

    return [
        'name' => $var,
        'label' => 'Block: Wysiwyg',
        'sub_fields' => [
            [
                'label' => __('Content', 'flynt'),
                'name' => 'contentHtml',
                'type' => 'wysiwyg',
                'delay' => 1,
                'media_upload' => 0,
                'required' => 1,
            ],
        ]
    ];
}

function foo() {
    global $var;

    echo $var;
}
```

`$var` in `getACFLayout()` and `foo()` returns `null` 😢

Thanks in advance!

**Comment by domtra** [comment, marked as answer]
Hi @wanjapflueger, 
I was  actually a bit surprised, when I saw your question. I would expected it to actually work exactly like you tried it.
However, after a bit of digging around, I found the reason why it is not working as expected.
We `require_once` the components' `function.php` files inside other functions inside WordPress hooks. So your initial `$var ='foo';` does actually not set a global variable.
In order to explicitly achieve what you are trying to, change the beginning of you code to 
```php
<?php

namespace Flynt\Components\BlockWysiwyg;

global $var;
$var = "foo";

function getACFLayout()
```

Hope that helps!

**Reply by wanjakarl** [reply]
Hi @domtra,

thank you very much for your reply!

👍 The suggested solution does the trick!  

👎 It does however interfere with intellisense and linting, since its a dirty solution. Please consider fixing in an upcoming release.

⚠️ Warning: You want to watch out for already declared global variable names (for example `$c` which is an _integer_ in _/wp-admin/admin.php_), so its best you choose a unique variable name that _Flynt_ or _WordPress_ does not already use.

This is now the working solution:

```php
<?php

namespace Flynt\Components\BlockWysiwyg;

global $uniqueUnicorn;
$uniqueUnicorn = "foo";

function getACFLayout()
{
    global $var;

    return [
        'name' => $uniqueUnicorn,
        // ...
    ];
}
```
